http-verbs-common/src/main/scala/uk/gov/hmrc/http/client2/package.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.http
+
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+
+package client2 {
+  trait Streaming
+}
+package object client2
+  extends StreamHttpReadsInstances {
+
+  // ensures strict HttpReads are not passed to stream function, which would lead to stream being read into memory
+  // (or runtime exceptions since HttpResponse.body with throw exception for streamed responses)
+  type StreamHttpReads[A] = HttpReads[A] with Streaming
```

Comment:
Worth sticking an `@implicitNotFound` hint on this?

Reply:
An implicit implementation is also provided here too

Reply:
Added a hint now since the provided implementation requires a Materializer